packages/context/src/binding.ts

```diff
@@ -227,19 +269,32 @@ export class Binding {
       }
     }
     if (this._getValue) {
-      const result = this._getValue(ctx);
+      const resolutionSession = ResolutionSession.enterBinding(this, session);
+      let result = RejectionError.catch(this._getValue(ctx, resolutionSession));
+      result = RejectionError.then<BoundValue, BoundValue>(result, val => {
```

Comment:
I see two important problems in this approach:

 1. By using method name `then` in `RejectionError.then`, it creates impression that `then` will behave as a regular Promise `then` (see the spec - https://promisesaplus.com/). AFAICT, this is not true. As a result, the timing order in which then-callback and code after `then()` is executed will be different from regular Promises, which can create subtle bugs.

 2. One of the goals of loopback-next was to avoid callbacks and direct Promise chaining, and use `async`, `await` and `try`/`catch` instead. This line and several places calling `RejectionError.then` are bringing us back to this legacy coding style.

As I commented before, can we come up with a better design that would not require `RejectionError` propagated as a resolved promise please?

Reply:
See my previous comment about why rejected promise cannot be propagated gracefully over async chain.